### PR TITLE
Use PacketField as a reference

### LINKED ISSUE #1714: Subpackage doesn't get newly initialised after previous initialisation
**\[edited: please remember to format code using triple \`\]**
I have written an example class and example unit test class to demonstrate the problem.

**class implementation:**
```python
from scapy.packet import *
from scapy.fields import *


class _INNER_CLASS(Packet):
    name = "InnerClass"
    fields_desc = [
        ShortField("id", 0),
        StrField("pkg_name", 'Xaver')
    ]

    def extract_padding(self, s):
        return "", s


class TESTCLASS(Packet):

    name = "testclass"

    fields_desc = [
        PacketField("innerclass", _INNER_CLASS(),_INNER_CLASS)
    ]
```

**unit test implementation:**
```
+ Testing

= load module

load_contrib("automotive.test")


= test1

p = TESTCLASS().innerclass
p.id = 5
p.pkg_name = "Peter"

p.show()

assert (p.id == 5)

= test2




p = TESTCLASS()
print(p.innerclass.id)
print(p.innerclass.pkg_name)

p.show()

assert(p.innerclass.pkg_name == "Xaver")
```

**output:**
```
###(000)=[passed] load module

>>> load_contrib("automotive.test")



###(001)=[passed] test1

>>> p = TESTCLASS().innerclass
>>> p.id = 5
>>> p.pkg_name = "Peter"
>>> 
>>> p.show()
###[ InnerClass ]### 
  id= 5
  pkg_name= 'Peter'

>>> 
>>> assert (p.id == 5)


###(002)=[failed] test2

>>> p = TESTCLASS()
>>> print(p.innerclass.id)
5
>>> print(p.innerclass.pkg_name)
Peter
>>> 
>>> p.show()
###[ testclass ]### 
  \innerclass\
   |###[ InnerClass ]### 
   |  id= 5
   |  pkg_name= 'Peter'

>>> 
>>> assert(p.innerclass.pkg_name == "Xaver")
Traceback (most recent call last):
  File "<input>", line 2, in <module>
AssertionError
```

As can be seen in the above output, the innerclass doesn't get newly initialised and still keeps the values it got before. Did I do something wrong here or  is this the default and wanted behaviour?

### PR DESCRIPTION
**Pull request by guedou**
This PR fixes #1714

### CODE REVIEW

File: scapy/packet.py [hunk 1]
```diff
@@ -240,7 +240,7 @@ def prepare_cached_fields(self, flist):
 
             # Remember references
             if isinstance(f.default, (list, dict, set)) or \
-                    isinstance(f.default, RandField):
+                    isinstance(f.default, (RandField, Packet)):
```
Comment (p-l-) [inline scapy/packet.py]: MMhhh this should be enough, or am I missing something? (not introduced by this commit, but could be fixed here)
```Python
             if isinstance(f.default, (list, dict, set, RandField, Packet)):
```

**Comment by guedou** [comment]
I can merge the two isinstance call. I intentionally wrote two to easily identify native Python types and Scapy ones.

**Comment by p-l-** [comment]
@guedou I'd rather have a single call to `isinstance()`, yes, if that's OK for you.

**Comment by guedou** [comment]
@p-l- done.

**Comment by p-l-** [comment]
@guedou Thanks!

**Pull request merged**